packages/picasso.js/src/core/dock-layout/dock-layout.js

```diff
@@ -308,10 +359,13 @@ export default function dockLayout(initialSettings) {
     validateComponent(component);
     docker.removeComponent(component);
 
+    const dock = component.dockConfig().dock();
+
     components.push({
       instance: component,
       key,
-      config: component.dockConfig()
+      config: component.dockConfig(),
+      referencedDocks: /^@/.test(dock) ? dock.split(',').map(s => s.replace(/^\s*@/, '')) : []
```

Comment:
this won't work if the first dock isn't a reference, e.g.: `left,@points`

Reply:
Good catch! Will fix! :)